stock_picking_group_by_max_weight/models/stock_move.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from odoo.osv.expression import AND
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    def _search_picking_for_assignation_domain(self):
+        """
+        Group pickings per customer
+        """
+        domain = super()._search_picking_for_assignation_domain()
+        if self.picking_type_id.group_pickings_maxweight:
+            domain = AND(
+                [
+                    domain,
+                    [("weight", "<=", self.picking_type_id.group_pickings_maxweight)],
```

Comment:
@rousseldenis The `weight` of the picking candidates should not make the current weight exceed the `group_pickings_maxweight`
```suggestion
                    [("weight", "<=", self.picking_type_id.group_pickings_maxweight - self.picking.weight)],
```

Reply:
In fact, the field evaluated on the picking should be the authorized maximum weight. So we needed to store that.